index.js

```diff
@@ -11,13 +13,134 @@ const FILE = 'pokedex.json';
+
+// this line below, sets a layout look to your express project
+const reactEngine = require('express-react-views').createEngine();
+app.engine('jsx', reactEngine);
+
+// this tells express where to look for the view files
+app.set('views', __dirname + '/views');
+
+// this line sets react to be the default view engine
+app.set('view engine', 'jsx');
+
+// app.get('/', (req, res) => {
+//   // running this will let express to run home.handlebars file in your views folder
+//   res.render('home')
+// })
+
 
 /**
  * ===================================
  * Routes
  * ===================================
  */
 
+app.get('/', (request, response)=> {
+    const nameArr = [];
+    jsonfile.readFile(FILE, (err, obj) => {
+        for (i=0; obj['pokemon'].length; i++) {
+            nameArr.push(obj['pokemon'][i]['name']);
+        }
+        const nameArrJoin = nameArr.join(", ");
+        response.send('<html><h1>Welcome to the online Pokedex!</h1><body>' + nameArrJoin + '<br>' +
+            '<form method="GET" action="/sortby = name"><input type="submit" value="Sort"></form></body></html>' )
+    })
+    console.log(nameArr);
+})
+
+
+app.get('/sortby=name', (request, response)=> {
+    nameArr.sort();
+    // console.log("request QUERY", request.query);
+})
+
+app.get('/pokemon/new', (request, response)=>{
+    let form = '<h1>New Pokemon</h1>'+
+                    '<form method="POST" action="/pokemon">' +
+                    'Make your Pokemon:' + '<br>' +
+                    'Pokemon ID: <input type="text" name="id">' + '<br>' +
+                    'Pokemon Number: <input type="text" name="num">' + '<br>' +
+                    'Pokemon Name: <input type="text" name="name">' + '<br>' +
+                    'Pokemon Image: <input type="text" name="img">' + '<br>' +
+                    'Height: <input type="text" name="height">' + '<br>' +
+                    'Weight: <input type="text" name="weight">' + '<br>' +
+                    '<input type="submit" value="Submit">' + '<br>' +
+                    '</form>';
+        response.send(form);
+});
+
+app.post('/pokemon',(request,response)=> {
+    console.log(request.body);
+
+    let submitForm = '<div> <h1>Pokemon</h1>' +
+                        '<div>' + request.body.id + '</div>' +
+                    '<div>' + request.body.num + '</div>' +
+                    '<div>' + request.body.name + '</div>' +
+                    '<div>' + request.body.height + '</div>' +
+                    '<div>' + request.body.weight + '</div></div>';
+    response.send(submitForm);
+
+    jsonfile.readFile(FILE, (err,obj)=>{
+        let newPokemon =[];
+
+        newPokemon.id = request.body.id;
+        newPokemon.num = request.body.num;
+        newPokemon.name = request.body.name;
+        newPokemon.img = request.body.img;
+        newPokemon.height = request.body.height;
+        newPokemon.weight = request.body.weight;
+
+            obj.pokemon.push(newPokemon);
```

Comment:
```suggestion
            obj["pokemon"].push(request.body);
```
don't need to create a new object since we are taking all the values of the given object in request.body anyway